src/arithmetic/bound_deducer.cc

```diff
@@ -128,7 +128,14 @@ class BoundDeducer: public IRVisitor {
     }
 
     // always use relax bound
-    result = result / operand + (is_greater ? 1 : -1);
+    Expr div_result = result / operand;
+    bool divided = can_prove(div_result * operand == result);
+    result = div_result;
+    if (is_greater) {
+      result += (divided && is_equal) ? 0 : 1;
```

Comment:
This piece of code is cryptic, need proper comment on why do we need repeatitive is_equal checks